cpp/src/arrow/csv/writer.cc

```diff
@@ -170,11 +178,17 @@ class UnquotedColumnPopulator : public ColumnPopulator {
       RETURN_NOT_OK(CheckStringArrayHasNoStructuralChars(*casted_array_, delimiter_));
     }
 
-    for (int x = 0; x < casted_array_->length(); x++) {
-      row_lengths[x] += casted_array_->IsNull(x)
-                            ? static_cast<int32_t>(null_string_->size())
-                            : casted_array_->value_length(x);
-    }
```

Comment:
`VisistArrayDataInline` processes continuous non-null / null blocks in batch and performs better than this bit by bit checking, though looks doing more work (extracts the string_view but only string length is needed).

Reply:
Nice fix, not sure why I didn't use this initially.